packages/ui/src/components/QueryBuilder/QueryBar.tsx

```diff
@@ -2,69 +2,76 @@ import React, { useEffect, useState } from 'react';
 import { AiOutlineCopy, AiOutlineDelete } from 'react-icons/ai';
 import { Button, Checkbox, Popconfirm } from 'antd';
 import cx from 'classnames';
-import isEmpty from 'lodash/isEmpty';
 
 import StackLayout from '../../layout/StackLayout';
 
-import Combiner from './Combiner';
-import QueryPill from './QueryPill';
-import { IDictionary, ISqonGroupFilter, TCallbackRemoveAction, TOnChange, TSqonGroupOp } from './types';
+import { IDictionary, TCallbackRemoveAction, TCallbackRemoveReferenceAction, TOnChange } from './types';
+import { ISyntheticSqon, TSqonGroupOp } from '../../data/sqon/types';
+import { isBooleanOperator, isEmptySqon } from '../../data/sqon/utils';
+import BooleanQueryPill from './QueryPills/BooleanQueryPill';
```

Comment:
why is it a boolean query pill?

Reply:
3 types:

Boolean Query Pill:
- Operator : 'and', 'or' or 'not'
- Combine 1 or more: Boolean Query Pill, Field Query Pill or Reference Query Pill

Field Query Pill
- Operator: '>', '>=', '<', etc.
- Combine a search field with different values

Reference Query Pill:
- No operator
- Reference another query